ApeWorX/ape
# feat: start ephemeral geth when no http provider active

### What I did

fixes: #204

### How I did it

Using `py-geth`, stand up an epheremal geth when using the http provider and they do not have a provider running at that URI.

### How to verify it

Deploy a script to the ephemeral geth.

First, you need to create an account.

```bash
ape account import geth_dev_0
```
And supply `0xdd23ca549a97cb330b011aebb674730df8b14acaee42d211ab45692699ab8ba5`
your address should be '0x1e59ce931B4CFea3fe4B875411e280e173cB7A9C'

Next, deploy a simple contract:

Contract:

```solidity
// SPDX-License-Identifier: MIT
pragma solidity ^0.6.0;

contract MyNumber {
    int public number = 42;
}
```

Script (get_my_number.py)

```python
from ape import accounts
from ape import project


def main():
    account = accounts.load("get_dev_0")

    contract_type = project.MyNumber
    contract = test_account.deploy(contract_type)

    my_num = contract.number(sender=test_account)
    print(my_num)

```

When you run the script:

```bash
ape run get_my_number
```

Follow the prompt to unlick your account.
It should print the number `42`.

Before that, it will have compiled the contracts if it needed as well as start up an ephemeral geth.

I can open the Testing PR after this that includes filtering out test accounts etc.


### Checklist
